src/base/NekRSProblem.C

```diff
@@ -694,20 +851,23 @@ NekRSProblem::syncSolutions(ExternalProblem::Direction direction)
       if (_volume && _has_heat_source)
         sendVolumeHeatSourceToNek();
 
-      // copy the boundary heat flux and/or volume heat source in the scratch space to device
-      nekrs::copyScratchToDevice(_minimum_scratch_size_for_coupling);
 
-      if (_moving_mesh)
+      if (nekrs::hasUserMeshSolver() && _volume)
```

Comment:
Is this boolean here actually equivalent to `if (nekrs::hasUserMeshSolver())`?

Reply:
We also want to ensure a volume mesh mirror exists to transfer displacements for all elements, including the interior ones. If the user has `user` mesh solver but just a boundary mesh mirror, we want to stop that and trigger an error (`nek_errors/deformation/missing_volume_for_user_solver`).
The `elasticity` solver works with both `_volume` and `_boundary` so there's no such check for `sendBoundaryDeformationToNek()`

Reply:
Sorry, I wasn't super precise. What I mean is that, earlier we throw an error if `_volume` is false, but the mesh solver is `user`. So by the time we get to this line, this two-term if-statement is redundant, right? Or am I missing some logic?